spec/unit/rust-crypto/rust-crypto.spec.ts

```diff
@@ -1879,6 +1881,79 @@ describe("RustCrypto", () => {
             );
         });
     });
+
+    describe("resetEncryption", () => {
+        let secretStorage: ServerSideSecretStorage;
+        beforeEach(() => {
+            secretStorage = {
+                setDefaultKeyId: jest.fn(),
+                hasKey: jest.fn().mockResolvedValue(false),
+                getKey: jest.fn().mockResolvedValue(null),
+            } as unknown as ServerSideSecretStorage;
+
+            fetchMock.post("path:/_matrix/client/v3/keys/upload", { one_time_key_counts: {} });
+            fetchMock.post("path:/_matrix/client/v3/keys/signatures/upload", {});
+        });
+
+        it("key backup should stay disabled after reset", async () => {
+            // We don't have a key backup
+            fetchMock.get("path:/_matrix/client/v3/room_keys/version", {});
+
+            const rustCrypto = await makeTestRustCrypto(makeMatrixHttpApi(), undefined, undefined, secretStorage);
+
+            const authUploadDeviceSigningKeys = jest.fn();
+            await rustCrypto.resetEncryption(authUploadDeviceSigningKeys);
+
+            // The default key id should be deleted
+            expect(secretStorage.setDefaultKeyId).toHaveBeenCalledWith(null);
+            expect(await rustCrypto.getActiveSessionBackupVersion()).toBeNull();
+            // The new cross signing keys should be uploaded
+            expect(authUploadDeviceSigningKeys).toHaveBeenCalledWith(expect.any(Function));
+        });
+
+        it("key backup should be re-enabled after reset", async () => {
+            fetchMock.get("path:/_matrix/client/v3/room_keys/version", testData.SIGNED_BACKUP_DATA);
+            // When we will delete the key backup
+            fetchMock.delete("path:/_matrix/client/v3/room_keys/version/1", {});
+
+            // We consider the key backup as trusted
+            jest.spyOn(RustBackupManager.prototype, "isKeyBackupTrusted").mockResolvedValue({
+                trusted: true,
+                matchesDecryptionKey: true,
+            });
```

Comment:
is this used before or after the reset?

Can we not use the actual backup data in the response to `/_matrix/client/v3/room_keys/version` so we don't need to mock this?

Reply:
It's used before the reset in order to have a trusted backup.

I suppose there is a better way to have a trusted backup returned by `/_matrix/client/v3/room_keys/version`?